.github/workflows/CI.yml

```diff
@@ -0,0 +1,113 @@
+name: CI
+
+on:
+  push:
+    paths-ignore:
+      - '.github/*'
+      - '.github/*_TEMPLATE/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - '.github/*'
+      - '.github/*_TEMPLATE/**'
+      - '*.md'
+
+jobs:
+  build-windows:
+    runs-on: windows-latest
+    env:
+      POWERSHELL_TELEMETRY_OPTOUT: 1
+      VALVE_NO_AUTO_P4: true
+      PreferredToolArchitecture: x64
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Release, Debug]
+        include:
+          - configuration: Release
+            creategameprojects: .\creategameprojects_dev.bat
+          - configuration: Debug
+            creategameprojects: .\creategameprojects_debug.bat
+    steps:
+      - uses: actions/checkout@v2
+      - uses: microsoft/setup-msbuild@v1.0.1
+      - name: Cache downloaded libs
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+            lib\common\*\2015\*\cryptlib.lib
+            lib\public\dme_controls.lib
+          key: ${{ runner.os }}
+      - name: Build libprotobuf
+        working-directory: thirdparty\protobuf-2.5.0\vsprojects
+        run: |
+          msbuild libprotobuf.vcxproj /p:Configuration=Release
+          msbuild libprotobuf.vcxproj /p:Configuration=Debug
+      - name: Download libs
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: .\download_libs.bat
+      - name: Create game projects
+        run: |
+          reg add "HKLM\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\10.0\Projects\{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}" /v DefaultProjectExtension /t REG_SZ /d vcproj /f
+          ${{ matrix.creategameprojects }}
+          Start-Process "${env:PROGRAMFILES(X86)}\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\devenv.com" "games.sln /upgrade" -nnw -Wait
```

Comment:
What does this upgrade line do? Seems like the CI output says it isn't needed?

Reply:
It makes the sln work with msbuild: #318

Reply:
Can we cache vcxproj, vpc and vpc.sentinel then?

Reply:
the vpc crc and sentinel will fail the check; vpc also adds a project dir preprocessor to the projects, you could cache a vpc-less vcxproj and it'll work but manual maintenance will be needed.
I believe reasons like those is why it's a good idea to either have VS2015 installed or add later VS support on VPC natively.

Reply:
Depends on when it is cached.

Reply:
What Momentum Mod does is add to the sln:
https://github.com/momentum-mod/game/blob/develop/mp/src/sln_fix.txt
https://github.com/momentum-mod/game/blob/develop/.github/workflows/CI.yml#L27

This is much faster than upgrading the sln.